modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManagerSelfTest.java

```diff
@@ -0,0 +1,880 @@
+        IgniteEx ig = startGridWithCache(defaultCacheCfg, CACHE_KEYS_RANGE);
+
+        Map<Integer, Set<Integer>> parts = new HashMap<>();
+        parts.put(CU.cacheId(DEFAULT_CACHE_NAME), new HashSet<>(Collections.singletonList(0)));
+
+        GridCacheSharedContext<?, ?> cctx0 = ig.context().cache().context();
+
+        IgniteInternalFuture<?> fut = startLocalSnapshotTask(cctx0.snapshotMgr(),
+            (GridCacheDatabaseSharedManager)cctx0.database(),
+            cctx0.localNodeId(),
+            SNAPSHOT_NAME,
+            parts,
+            new DeleagateSnapshotSender(log, cctx0.snapshotMgr().snapshotExecutorService(),
+                cctx0.snapshotMgr().localSnapshotSender(SNAPSHOT_NAME)) {
+                @Override public void sendPart0(File part, String cacheDirName, GroupPartitionId pair, Long length) {
+                    if (pair.getPartitionId() == 0)
+                        throw new IgniteException("Test. Fail to copy partition: " + pair);
+
+                    delegate.sendPart0(part, cacheDirName, pair, length);
+                }
+            });
+
+        fut.get();
+    }
+
+    /**
+     * @throws Exception If fails.
+     */
+    @Test
+    public void testSnapshotRemotePartitions() throws Exception {
+        IgniteEx ig0 = startGrids(2);
+
+        ig0.cluster().active(true);
+
+        for (int i = 0; i < CACHE_KEYS_RANGE; i++)
+            ig0.cache(DEFAULT_CACHE_NAME).put(i, i);
+
+        CheckpointProgress cpFut = ig0.context()
+            .cache()
+            .context()
+            .database()
+            .forceCheckpoint("the next one");
+
+        cpFut.futureFor(CheckpointState.FINISHED).get();
+
+        IgniteSnapshotManager mgr0 = ig0.context()
+            .cache()
+            .context()
+            .snapshotMgr();
+
+        final CountDownLatch cancelLatch = new CountDownLatch(1);
+
+        UUID rmtNodeId = grid(1).localNode().id();
+
+        // Snapshot must be taken on node1 and transmitted to node0.
+        IgniteInternalFuture<?> fut = mgr0.createRemoteSnapshot(rmtNodeId,
+            owningParts(ig0, new HashSet<>(Collections.singletonList(CU.cacheId(DEFAULT_CACHE_NAME))), rmtNodeId),
+            new BiConsumer<File, GroupPartitionId>() {
+                @Override public void accept(File file, GroupPartitionId gprPartId) {
+                    log.info("Snapshot partition received successfully [rmtNodeId=" + rmtNodeId +
+                        ", part=" + file.getAbsolutePath() + ", gprPartId=" + gprPartId + ']');
+
+                    cancelLatch.countDown();
+                }
+            });
+
+        cancelLatch.await();
+
+        fut.cancel();
+
+        IgniteInternalFuture<?> fut2 = mgr0.createRemoteSnapshot(rmtNodeId,
+            owningParts(ig0, new HashSet<>(Collections.singletonList(CU.cacheId(DEFAULT_CACHE_NAME))), rmtNodeId),
+            (part, grp) -> {});
+
+        fut2.get();
+    }
```

Comment:
What do we check in this test? There are no assertions in the test and no comments to help to understand what conditions are we checking.